Lib/test/test_urllib.py

```diff
@@ -426,6 +426,22 @@ def test_ftp_nonexisting(self):
         self.assertFalse(e.exception.filename)
         self.assertTrue(e.exception.reason)
 
+    def test_ftp_illegalhost(self):
+        illegal_ftp_hosts = [
+            'ftp://foo:bar%0d%0aINJECTED@example.net/file.png',
+            'fTp://foo:bar%0d%0aINJECTED@example.net/file.png',
+            'FTP://foo:bar%0d%0aINJECTED@example.net/file.png',
+            'ftp://foo:bar%0aINJECTED@example.net/file.png',
+            'fTp://foo:bar%0aINJECTED@example.net/file.png',
+            'FTP://foo:bar%0aINJECTED@example.net/file.png'
```

Comment:
I don't think that it's interesting to test different cases, I suggest to only test %0d%0a and %0a, so only keep two tests using ftp:// prefix.